### PR TITLE
made redux changes

### PR DESCRIPTION
**Pull request by nulfrost**
updating the state works now, just getting the data into it is where the problem is now

### CODE REVIEW

File: src/ExpenseItemList.tsx [hunk 1]
```diff
@@ -1,39 +1,36 @@
 import React, { ChangeEvent } from 'react';
-import ExpenseItem from "./ExpenseItem";
+import ExpenseItem from './ExpenseItem';
```
Comment (usrrname) [inline src/ExpenseItemList.tsx:2]: Seems like our linters let different things fly.

Comment (nulfrost) [reply src/ExpenseItemList.tsx:2]: double quotes are the devil

File: src/store/reducers/ItemListReducer.ts [hunk 2]
```diff
@@ -42,51 +45,57 @@ export const DELETE_ITEM = (id: string): Action => {
...
-			return state
-	}
-}
+export const ItemListReducer: Reducer<IState, Action> = (
+  state = initialState,
+  action: ListActions
+) => {
+  switch (action.type) {
+    // here's what i changed that made it work, my initial thought was that the name you used for these cases didn't match up
+    // with the action names you were dispatching. the code for updating the state was right
```
Comment (usrrname) [inline src/store/reducers/ItemListReducer.ts:54]: You're spot on with that... the `{$template literals}` were really extra.

File: src/ExpenseItemList.tsx [hunk 3]
```diff
@@ -1,39 +1,36 @@
...
-  items: Item[],
-  onChange?: (event: ChangeEvent<HTMLInputElement>) => void,
-  deleteItem: (id: string) => void
-}
+  items: Item[];
+  onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
+  deleteItem: (id: string) => void;
+};
 
+// adding new items comes up as null for some reason but the state gets updated
```
Comment (usrrname) [inline src/ExpenseItemList.tsx:11]: I see what you're saying now. Totally confounding!

File: src/store/reducers/ItemListReducer.ts [hunk 4]
```diff
@@ -42,51 +45,53 @@ export const DELETE_ITEM = (id: string): Action => {
...
-}
+  let newItem: Item = {
+    id: v4(),
+    name: '',
+    cost: 0
+  };
+  console.log('test');
+  return (dispatch, getState) => {
+    // fixed it! was missing 'payload' lol
+    dispatch({ type: ItemActions.ADD_ITEM, payload: newItem });
```
Comment (usrrname) [inline src/store/reducers/ItemListReducer.ts:82]: Holy! Wow thank you! 👏  👏 👏 
One can not simply assume a param name 🤣

**Review by usrrname — APPROVED** [review]

File: src/store/reducers/ItemListReducer.ts (hunk 4 above)
Comment (nulfrost) [reply src/store/reducers/ItemListReducer.ts:82]: haha no problem :)

**Pull request merged**